File: AI-agent-backend-fastapi/app/models/test_data.py
"""
Copyright (c) 2025 左岚. All rights reserved.
测试数据模型
"""
from datetime import datetime
from sqlalchemy import Column, Integer, String, Text, DateTime
from app.core.database import Base


class TestData(Base):
    """测试数据模型"""

    __tablename__ = "test_data"

    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    name = Column(String(200), nullable=False, comment="数据名称")
    data_type = Column(String(50), nullable=False, default="json", comment="数据类型: json, csv, text, sql")
    description = Column(Text, comment="描述")
    content = Column(Text, nullable=False, comment="数据内容")
    created_at = Column(DateTime, default=datetime.utcnow, comment="创建时间")
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, comment="更新时间")

    def to_dict(self):
        """转换为字典"""
        return {
            "id": self.id,
            "name": self.name,
            "data_type": self.data_type,
            "description": self.description,
            "content": self.content,
            "created_at": self.created_at.strftime("%Y-%m-%d %H:%M:%S") if self.created_at else None,
            "updated_at": self.updated_at.strftime("%Y-%m-%d %H:%M:%S") if self.updated_at else None,
        }

File: AI-agent-frontend-vue3/src/api/test-data.ts
// Copyright (c) 2025 左岚. All rights reserved.
import request from './request'

export interface TestData {
    id?: number
    name: string
    data_type: 'json' | 'csv' | 'text' | 'sql'
    description?: string
    content: string
    created_at?: string
    updated_at?: string
}

export interface TestDataListParams {
    keyword?: string
    data_type?: string
    page?: number
    page_size?: number
}

/**
 * 获取测试数据列表
 */
export const getTestDataListAPI = (params: TestDataListParams) => {
    return request({
        url: '/api/v1/test-data',
        method: 'get',
        params
    })
}

/**
 * 获取测试数据详情
 */
export const getTestDataDetailAPI = (id: number) => {
    return request({
        url: `/api/v1/test-data/${id}`,
        method: 'get'
    })
}

/**
 * 创建测试数据
 */
export const createTestDataAPI = (data: TestData) => {
    return request({
        url: '/api/v1/test-data',
        method: 'post',
        data
    })
}

/**
 * 更新测试数据
 */
export const updateTestDataAPI = (id: number, data: Partial<TestData>) => {
    return request({
        url: `/api/v1/test-data/${id}`,
        method: 'put',
        data
    })
}

/**
 * 删除测试数据
 */
export const deleteTestDataAPI = (id: number) => {
    return request({
        url: `/api/v1/test-data/${id}`,
        method: 'delete'
    })
}

